src/lib.rs

```diff
@@ -1,253 +1,253 @@
 
 struct TaskMonitor {
-  counter: AtomicUsize,
-  tasks: BTreeSet<TaskMonitorId>,
-  progress_bar: Arc<indicatif::ProgressBar>,
+    counter: usize,
+    tasks: BTreeSet<TaskMonitorId>,
+    progress_bar: indicatif::ProgressBar,
 }
 
 impl TaskMonitor {
-  fn new(progress_bar: Arc<indicatif::ProgressBar>) -> TaskMonitor {
-    TaskMonitor {
-      counter: AtomicUsize::new(0),
-      tasks: BTreeSet::new(),
-      progress_bar,
+    fn new(progress_bar: indicatif::ProgressBar) -> TaskMonitor {
+        TaskMonitor {
+            counter: 0,
+            tasks: BTreeSet::new(),
+            progress_bar,
+        }
+    }
+
+    fn started(&mut self, task_name: String) -> TaskMonitorId {
+        let now = Instant::now();
+        let task_id = TaskMonitorId((now, task_name, self.counter));
+        self.counter += 1;
+        self.tasks.insert(task_id.clone());
+        self.update_progress_bar();
+        task_id
     }
-  }
-
-  fn started(&mut self, task_name: String) -> TaskMonitorId {
-    let now = Instant::now();
-    let counter = self.counter.fetch_add(1, Ordering::Relaxed);
-    let task_id = TaskMonitorId((now, task_name, counter));
-    self.tasks.insert(task_id.clone());
-    self.update_progress_bar();
-    task_id
-  }
-
-  fn finished(&mut self, task_id: TaskMonitorId) {
-    let removed = self.tasks.remove(&task_id);
-    assert!(removed);
-    self.progress_bar.inc(1);
-    self.update_progress_bar();
-  }
-
-  fn update_progress_bar(&mut self) {
-    match self.tasks.iter().next() {
-      Some(task) => {
-        self.progress_bar.set_message((task.0).1.clone());
-      }
-
-      None => {
-        self.progress_bar.set_message("");
-      }
+
+    fn finished(&mut self, task_id: TaskMonitorId) {
+        let removed = self.tasks.remove(&task_id);
+        assert!(removed);
+        self.progress_bar.inc(1);
+        self.update_progress_bar();
+    }
+
+    fn update_progress_bar(&mut self) {
+        match self.tasks.iter().next() {
+            Some(task) => {
+                self.progress_bar.set_message((task.0).1.clone());
+            }
+
+            None => {
+                self.progress_bar.set_message("");
+            }
+        }
     }
-  }
 }
 
 pub struct Pool {
-  thread_count: Option<usize>,
-  thread_pool: Option<futures::executor::ThreadPool>,
-  quiet: bool,
+    thread_count: Option<usize>,
+    thread_pool: Option<rayon::ThreadPool>,
+    quiet: bool,
 }
 
 pub struct ExecutionResult<T> {
-  pub name: String,
-  pub result: T,
+    pub name: String,
+    pub result: T,
 }
 
 pub struct ExecutionResults<T, E> {
-  pub successful: Vec<ExecutionResult<T>>,
-  pub failed: Vec<ExecutionResult<E>>,
+    pub successful: Vec<ExecutionResult<T>>,
+    pub failed: Vec<ExecutionResult<E>>,
 }
 
 impl Pool {
-  pub fn with_default_size() -> Pool {
-    Pool {
-      thread_count: None,
-      thread_pool: None,
-      quiet: false,
+    pub fn with_default_size() -> Pool {
```

Comment:
Should we refactor this into `Default` impl?

Reply:
I'm not sure that's better. This name is more descriptive than "Default".